yarn-project/acir-simulator/src/avm/opcodes/opcode.ts

```diff
@@ -0,0 +1,9 @@
+import { AvmContext } from '../avm_context.js';
+import { AvmStateManager } from '../avm_state_manager.js';
+
+/**
+ * Opcode base class
+ */
+export abstract class Opcode {
+  abstract execute(context: AvmContext, stateManager: AvmStateManager): void;
+}
```

Comment:
Should this really be `Instruction` or `AvmInstruction`? An opcode is just the first few bits of an instruction.

If you have `Add`, for example, which implements this abstract class, it can be instantiated with flags & arguments. Your instance of `Add` with its arguments is really an instruction that can later be executed. 

Thoughts?

Reply:
Good point, im stuck in the evm nomenclature - will update